engines/alg/logic/game_bountyhunter.cpp

```diff
@@ -0,0 +1,1353 @@
+			}
+			if (_videoDecoder->getCurrentFrame() == 0) {
+				_videoDecoder->getNextFrame();
+			}
+			updateScreen();
+			int32 remainingMillis = _nextFrameTime - getMsTime();
+			if (remainingMillis < 10) {
+				if (_videoDecoder->getCurrentFrame() > 0) {
+					_videoDecoder->getNextFrame();
+				}
+				remainingMillis = _nextFrameTime - getMsTime();
+				_nextFrameTime = getMsTime() + (remainingMillis > 0 ? remainingMillis : 0) + 100;
+			}
+			if (remainingMillis > 0) {
+				if (remainingMillis > 15) {
+					remainingMillis = 15;
+				}
+				g_system->delayMillis(remainingMillis);
+			}
+			_currentFrame = getFrame(scene);
+		}
+		// frame limit reached or scene changed, prepare for next scene
+		_hadPause = false;
+		_pauseTime = 0;
+		if (_curScene == oldscene) {
+			callScriptFunctionScene(NXTSCN, scene->_nxtscn, scene);
+		}
+		if (_curScene == "") {
+			_vm->quitGame();
+		}
+	}
+	return Common::kNoError;
+}
+
+void GameBountyHunter::newGame() {
+	_playerLives[0] = _playerLives[1] = 3;
+	_playerShots[0] = _playerShots[1] = 6;
+	_playerGun[0] = _playerGun[1] = 1;
+	_playerScore[0] = _playerScore[1] = 0;
+	_currentSubLevelSceneId = 0x017B;
+	_holster = false;
+}
+
+void GameBountyHunter::doMenu() {
+	uint32 startTime = getMsTime();
+	updateCursor();
+	_inMenu = true;
+	moveMouse();
+	g_system->getMixer()->pauseHandle(_sceneAudioHandle, true);
+	_screen->copyRectToSurface(_background->getBasePtr(_videoPosX, _videoPosY), _background->pitch, _videoPosX, _videoPosY, _videoDecoder->getWidth(), _videoDecoder->getHeight());
+	while (_inMenu && !_vm->shouldQuit()) {
+		Common::Point firedCoords;
+		if (fired(&firedCoords)) {
+			Rect *hitMenuRect = checkZone(_submenzone, &firedCoords);
+			if (hitMenuRect != nullptr) {
+				callScriptFunctionRectHit(hitMenuRect->_rectHit, hitMenuRect);
+			}
+		}
+		updateScreen();
+		g_system->delayMillis(15);
+	}
+	updateCursor();
+	g_system->getMixer()->pauseHandle(_sceneAudioHandle, false);
+	if (_hadPause) {
+		uint32 endTime = getMsTime();
+		uint32 timeDiff = endTime - startTime;
+		_pauseTime += timeDiff;
+		_nextFrameTime += timeDiff;
+	}
+}
+
+void GameBountyHunter::updateCursor() {
+	updateMouse();
+}
+
+void GameBountyHunter::updateMouse() {
+	if (_oldWhichGun != _whichGun) {
+		Graphics::Surface *cursor = (*_gun)[_whichGun];
+		if (_playerGun[0] == 2 && _whichGun < 2) {
+			cursor = (*_shotgun)[_whichGun];
+		}
+		CursorMan.popAllCursors();
+		uint16 hotspotX = (cursor->w / 2) + 8;
+		uint16 hotspotY = (cursor->h / 2) + 8;
+		if (debugChannelSet(1, Alg::kAlgDebugGraphics)) {
+			cursor->drawLine(0, hotspotY, cursor->w, hotspotY, 1);
+			cursor->drawLine(hotspotX, 0, hotspotX, cursor->h, 1);
+		}
+		CursorMan.pushCursor(cursor->getPixels(), cursor->w, cursor->h, hotspotX, hotspotY, 0);
+		CursorMan.showMouse(true);
+		_oldWhichGun = _whichGun;
+	}
+}
+
+void GameBountyHunter::moveMouse() {
+	if (_inMenu) {
+		_whichGun = 3; // in menu cursor
+	} else {
+		// disabled for now, because glitchy
```

Comment:
```suggestion
		// TODO: disabled for now, because glitchy
```

This way you can easily grep for it

Reply:
done